phoenix-hbase-compat-2.3.0/pom.xml

```diff
@@ -97,6 +97,18 @@
       <version>${hbase23.compat.version}</version>
       <scope>provided</scope>
     </dependency>
+    <dependency>
+      <groupId>org.slf4j</groupId>
+      <artifactId>slf4j-api</artifactId>
+      <version>${slf4j.version}</version>
+      <scope>provided</scope>
+    </dependency>
+    <dependency>
```

Comment:
Could you solve this without Guava ?
I know that it doesn't really matter, but it pains me to see anther Guava dependency added.

Reply:
For 4.x `Uninterruptibles` is in beta version based on guava version that we use so i used this: https://github.com/apache/phoenix/pull/1097/files#diff-d12329d4796498b9880a54bb4b6d681f7ee42b924536bf5247791f372329cd9fR61
However, i feel using `Uninterruptibles.sleepUninterruptibly` more comfortably as we don't have to catch `InterruptedException` and it is not interruptible. But i agree with your point on not introducing more dependency so let me replicate same logic as 4.x.

Reply:
Done